### PR TITLE
[fix](tabletScheduler) Fix addTablet dead lock in tabletScheduler

### PR DESCRIPTION
**Pull request by deardeng**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

The conditions that need to be met to trigger the bug, with the second condition being somewhat difficult to trigger, are as follows:
1. The number of tablets that need to be fixed exceeds 2000 (in the pending queue);
2. The scheduling of the lowest priority in the pending queue has previously experienced a clone failure, with fewer than 3 failures, and has been put back into the pending queue. Additionally, a new scheduling request that happens to belong to the same table as the previous one has a higher priority than the previous scheduling.

The fix is to write the lock trylock in finalize TabletCtx. If the lock cannot be obtained, the current scheduling will fail and the next one will be rescheduled

Fix
```
"colocate group clone checker" #7557 daemon prio=5 os_prio=0 cpu=686.24ms elapsed=6719.45s tid=0x00007f3e6c039ab0 nid=0x17b08 waiting on condition  [0x00007f3ec77fe000]
(1 similar threads)
   java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.2/Native Method)
        - parking to wait for  <0x000010014d223908> (a java.util.concurrent.locks.ReentrantReadWriteLock$FairSync)
        at java.util.concurrent.locks.LockSupport.park(java.base@17.0.2/LockSupport.java:211)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.2/AbstractQueuedSynchronizer.java:715)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.2/AbstractQueuedSynchronizer.java:938)
        at java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock(java.base@17.0.2/ReentrantReadWriteLock.java:959)
        at org.apache.doris.common.lock.MonitoredReentrantReadWriteLock$WriteLock.lock(MonitoredReentrantReadWriteLock.java:98)
        at org.apache.doris.catalog.Table.writeLockIfExist(Table.java:211)
        at org.apache.doris.clone.TabletSchedCtx.releaseResource(TabletSchedCtx.java:940)
        at org.apache.doris.clone.TabletSchedCtx.releaseResource(TabletSchedCtx.java:898)
        at org.apache.doris.clone.TabletScheduler.releaseTabletCtx(TabletScheduler.java:1743)
        at org.apache.doris.clone.TabletScheduler.finalizeTabletCtx(TabletScheduler.java:1625)
        at org.apache.doris.clone.TabletScheduler.addTablet(TabletScheduler.java:287)
        - locked <0x0000100009429110> (a org.apache.doris.clone.TabletScheduler)
        at org.apache.doris.clone.ColocateTableCheckerAndBalancer.matchGroups(ColocateTableCheckerAndBalancer.java:563)
        at org.apache.doris.clone.ColocateTableCheckerAndBalancer.runAfterCatalogReady(ColocateTableCheckerAndBalancer.java:340)
        at org.apache.doris.common.util.MasterDaemon.runOneCycle(MasterDaemon.java:58)
        at org.apache.doris.common.util.Daemon.run(Daemon.java:119)
```

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by gavinchou — DISMISSED** [review]

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by yujun777** [comment]
LGTM

File: fe/fe-core/src/test/java/org/apache/doris/clone/TabletHealthTest.java [hunk 1]
```diff
@@ -61,6 +63,7 @@ protected void beforeCreatingConnectContext() throws Exception {
         Config.colocate_group_relocate_delay_second = -1000; // be dead will imm relocate
         Config.tablet_schedule_interval_ms = 7200_000L;  //disable schedule
         Config.tablet_checker_interval_ms = 7200_000L;  //disable checker
+        Config.max_scheduling_tablets = 2000;
```
Comment (yujun777) [inline fe/fe-core/src/test/java/org/apache/doris/clone/TabletHealthTest.java:66]: this only run once, move it to runBeforeEach() function

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by deardeng** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**